example.html

```diff
@@ -1,63 +1,112 @@
+        gui.add(obj, 'displayOutline');
+        gui.add(obj, 'explode');
 
-    gui.remember(obj);
+        gui.add(obj, 'maxSize').min(-10).max(10).step(0.25);
+        gui.add(obj, 'height').step(5); // Increment amount
 
-    gui.add(obj, 'message');
-    gui.add(obj, 'displayOutline');
-    gui.add(obj, 'explode');
+        // Choose from accepted values
+        gui.add(obj, 'type', ['one', 'two', 'three']);
 
-    gui.add(obj, 'maxSize').min(-10).max(10).step(0.25);
-    gui.add(obj, 'height').step(5); // Increment amount
+        // Choose from named values
+        gui.add(obj, 'speed', {
+            Stopped: 0,
+            Slow: 0.1,
+            Fast: 5
+        });
 
-    // Choose from accepted values
-    gui.add(obj, 'type', [ 'one', 'two', 'three' ] );
+        var f1 = gui.addFolder('Colors');
+        f1.addColor(obj, 'color0');
+        f1.addColor(obj, 'color1');
+        f1.addColor(obj, 'color2');
+        f1.addColor(obj, 'color3');
 
-    // Choose from named values
-    gui.add(obj, 'speed', { Stopped: 0, Slow: 0.1, Fast: 5 } );
+        var f2 = gui.addFolder('Another Folder');
+        f2.add(obj, 'noiseStrength');
 
-    var f1 = gui.addFolder('Colors');
-    f1.addColor(obj, 'color0');
-    f1.addColor(obj, 'color1');
-    f1.addColor(obj, 'color2');
-    f1.addColor(obj, 'color3');
+        var f3 = f2.addFolder('Nested Folder');
+        f3.add(obj, 'growthSpeed');
 
-    var f2 = gui.addFolder('Another Folder');
-    f2.add(obj, 'noiseStrength');
+        //  alternative StringController add
+        gui.add(new dat.controllers.StringController(obj, 'message'));
 
-    var f3 = f2.addFolder('Nested Folder');
-    f3.add(obj, 'growthSpeed');
+        //  a custom controller
+        class KnobController extends dat.controllers.Controller {
+            constructor(object, property, a, b) {
+                super(object, property);
+                const _this = this;
+                this.__input = document.createElement('input');
+                this.__input.setAttribute('type', 'number');
+                this.__input.style.width = '40%';
+                this.updateDisplay();
+                this.domElement.appendChild(this.__input);
+                var button = document.createElement('input');
+                button.setAttribute('type', 'button');
+                button.value = 'Set ' + property;
+                button.style.width = '50%';
+                button.onclick = function(e) {
+                    object[property] = a + b;
+                    _this.updateDisplay();
+                };
+                this.domElement.appendChild(button);
+            }
+            updateDisplay() {
+                this.__input.value = this.getValue();
+            }
+        }
 
-  </script>
+        const api = {
+
+            color: '#ffffff',
+            value: 0.5,
+
+        };
+
+        gui.add(new KnobController(api, 'value', 0.5, 25), {
+            liClass: 'knobby'
+        });
```

Comment:
Do you know if this is compatible with the `augmentController` method here?

https://github.com/dataarts/dat.gui/blob/096993eb6eecad0481b03c40862a11a4ab0db8e9/src/dat/gui/GUI.js#L922-L990

... specifically — can you still remove(), listen(), and name()  the KnobController normally? I don't suppose it's possible to override those methods in the subclass, which is unfortunate, but probably not this PR's problem to solve.